lib/archethic/db/embedded_impl/inputs_reader.ex

```diff
@@ -0,0 +1,34 @@
+defmodule Archethic.DB.EmbeddedImpl.InputsReader do
+  @moduledoc """
+  Inputs are stored by destination address. 1 file per address per ledger
+  """
+
+  alias Archethic.DB.EmbeddedImpl.InputsWriter
+  alias Archethic.TransactionChain.VersionedTransactionInput
+  alias Archethic.Utils
+
+  @spec get_inputs(ledger :: InputsWriter.ledger(), address :: binary()) ::
+          list(VersionedTransactionInput.t())
+  def get_inputs(ledger, address) do
+    filename = InputsWriter.address_to_filename(ledger, address)
+
+    case File.read(filename) do
+      {:error, :enoent} ->
+        []
+
+      {:ok, bin} ->
+        bin
+        |> deserialize_inputs_file([])
+    end
+  end
+
+  defp deserialize_inputs_file(<<>>, acc), do: Enum.reverse(acc)
```

Comment:
I think the `Enum.reverse` is not necessary as the write is not written in a specific order.